src/lib/io/sol-iio.c

```diff
@@ -1052,12 +1051,14 @@ sol_iio_read_channel_value(struct sol_iio_channel *channel, double *value)
     int len;
     int64_t raw_value;
     char path[PATH_MAX];
-    struct sol_iio_device *device = channel->device;
+    struct sol_iio_device *device;
     bool r;
 
     SOL_NULL_CHECK(channel, false);
     SOL_NULL_CHECK(value, false);
 
+    device = channel->device;
```

Comment:
For one suggestion, maybe we can just add one line to check SOL_NULL_CHECK(device, false) which would be better and safer, because for this issue, segment fault result from struct device being NULL. Of course, in this case that set_buffer_enable do not goto error, this segfault would never happen.